gui/src/components/Transactions/List.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Card from '@material-ui/core/Card'
+import Table from '@material-ui/core/Table'
+import TableBody from '@material-ui/core/TableBody'
+import TableCell from '@material-ui/core/TableCell'
+import TableHead from '@material-ui/core/TableHead'
+import TableRow from '@material-ui/core/TableRow'
+import TablePagination from '@material-ui/core/TablePagination'
+import Typography from '@material-ui/core/Typography'
+import Link from 'components/Link'
+import TableButtons, { FIRST_PAGE } from 'components/TableButtons'
+import { useHooks, useState, useEffect } from 'use-react-hooks'
+
+const renderBody = (transactions, error) => {
+  if (error) {
+    return (
+      <TableRow>
+        <TableCell component='th' scope='row' colSpan={3}>
+          {error}
+        </TableCell>
+      </TableRow>
+    )
+  } else if (transactions && transactions.length === 0) {
+    return (
+      <TableRow>
+        <TableCell component='th' scope='row' colSpan={3}>
+          You haven't created any transactions yet.
+        </TableCell>
+      </TableRow>
+    )
+  } else if (transactions) {
+    return transactions.map(j => (
+      <TableRow key={j.hash}>
+        <TableCell component='th' scope='row'>
+          <Link to={`/transactions/${j.hash}`}>{j.hash}</Link>
+        </TableCell>
+        <TableCell>
+          <Typography variant='body1'>{j.txId}</Typography>
+        </TableCell>
+      </TableRow>
+    ))
+  }
+
+  return (
+    <TableRow>
+      <TableCell component='th' scope='row' colSpan={3}>
+        Loading...
+      </TableCell>
+    </TableRow>
+  )
+}
+
+export const List = useHooks(props => {
+  const [ page, setPage ] = useState(FIRST_PAGE)
+  useEffect(() => {
+    const queryPage = (props.match && parseInt(props.match.params.transactionsPage, 10)) || FIRST_PAGE
+    setPage(queryPage)
+    fetchTransactions(queryPage, pageSize)
+  }, [])
+  const { transactions, count, fetchTransactions, pageSize, error } = props
+  const handleChangePage = (e, page) => {
+    fetchTransactions(page, pageSize)
+    setPage(page)
```

Comment:
Wouldn't we ideally want to response from `fetchTransactions` to `setPage(...)`? i.e. in the response handler of the promise to fetch transactions, it triggers a rerender of the component that has the right page number. I've noticed a few instances where we have the 

```
fetchTransactions(...)
setPage(...)
```

pair.

This way, you're doing two rerenders, one for an initially empty page that might show 
```
You haven't created any transactions yet.
```
rather than loading, which might flicker to have results.

Reply:
@dimroc `fetchTransactions` is changing the shape of redux state & `setPage` is changing the state local to the component